LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsO.cs

```diff
@@ -1047,6 +1035,918 @@ public static void CopyMatToBMOorMIC(PackageEditorWindow pew)
+                    return;
+                }
+            }
+
+            string oldBankName = wwiseBankEntry.ObjectName;
+            IntProperty bankIDProp = wwiseBankEntry.GetProperty<IntProperty>("Id");
+
+            // Get/Generate IDs and little endian hashes
+            uint oldBankID = unchecked((uint)bankIDProp.Value);
+            string oldBankHash = BigToLittleEndian(string.Format("{0:X2}", oldBankID).PadLeft(8, '0'));
+
+            uint newBankID = GetBankId(newWwiseBankName);
+            string newBankHash = BigToLittleEndian(string.Format("{0:X2}", newBankID).PadLeft(8, '0'));
+
+            // Write the replaced ID property
+            bankIDProp.Value = unchecked((int)newBankID);
+            wwiseBankEntry.WriteProperty(bankIDProp);
+
+            WwiseBank wwiseBank = wwiseBankEntry.GetBinaryData<WwiseBank>();
+            // Update the bank id
+            wwiseBank.ID = newBankID;
+
+            // Update referenced banks kvp that reference the old name
+            IEnumerable<KeyValuePair<uint, string>> updatedBanks = wwiseBank.ReferencedBanks
+                .Select(referencedBank =>
+                {
+                    if (referencedBank.Value.Equals(oldBankName, StringComparison.OrdinalIgnoreCase)) { return new(newBankID, newWwiseBankName); }
+                    else { return referencedBank; }
+                });
+            wwiseBank.ReferencedBanks = new OrderedMultiValueDictionary<uint, string>(updatedBanks);
+
+            // Update references to the old hash at the end of HIRC objects when they are found.
+            // This is mostly safe, since we know the reference appears at the end of the unparsed data,
+            // and we only replace it there.
+            foreach (WwiseBank.HIRCObject hirc in wwiseBank.HIRCObjects.Values())
+            {
+                byte[] unparsed = hirc.unparsed;
+                if (unparsed != null && unparsed.Length >= 4) // Only replace if not null and at least width of hash
+                {
+                    byte[] oldArr = Convert.FromHexString(oldBankHash);
+                    byte[] newArr = Convert.FromHexString(newBankHash);
+
+                    int idBase = unparsed.Length - 4;
+
+                    // Check if the last 4 bytes of unparsed match the old hash
+                    bool equal = true;
+                    for (int i = 0; i < 4; i++)
+                    {
+                        equal = equal && (unparsed[idBase + i] == oldArr[i]);
+                    }
+
+                    // Replace the hash
+                    if (equal)
+                    {
+                        for (int i = 0; i < 4; i++)
+                        {
+                            unparsed[idBase + i] = newArr[i];
+                        }
+
+                        hirc.unparsed = unparsed;
+                    }
```

Comment:
This entire section could be greatly simplified through usage of Spans.
```suggestion
                    Span<byte> last4Bytes = unparsed.AsSpan(^4..);

                    if (last4Bytes.SequenceEqual(oldArr))
                    {
                        newArr.CopyTo(last4Bytes);
                    }
```
Arrays are reference types, so `hirc.unparsed = unparsed` is a no-op, since `unparsed` was initialized from `hirc.unparsed` and never reassigned.

Reply:
I did not know about Spans, this is pretty neat